test/end-to-end/offer.js

```diff
@@ -1103,4 +1109,400 @@ describe("Offer endpoint tests", () => {
+
+            test("should fail if offer does not exist", async () => {
+                const _id = "111111111111111111111111";
+                const res = await test_agent
+                    .post(`/offers/edit/${_id}`)
+                    .send(withGodToken())
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", `Error: ${ValidationReasons.OFFER_NOT_FOUND(_id)}`);
+            });
+
+
+            test("offer with grace period over", async () => {
+
+                const res = await test_agent
+                    .post(`/offers/edit/${expired_test_offer._id.toString()}`)
+                    .send(withGodToken())
+                    .expect(HTTPStatus.FORBIDDEN);
+                expect(res.body).toHaveProperty("reason", ValidationReasons.OFFER_EDIT_PERIOD_OVER(expired_test_offer._id));
+            });
+
+            describe("should fail if offer with grace period over", () => {
+                const RealDateNow = Date.now;
+                const mockDate = new Date(Date.now() + (OFFER_EDIT_GRACE_PERIOD * HOUR_IN_MS * 2));
+                beforeEach(() => {
+                    Date.now = () => mockDate.getTime();
+                });
+
+                afterEach(() => {
+                    Date.now = RealDateNow;
+                });
+
+                test("should fail offer with grace period over", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${grace_period_over_test_offer._id.toString()}`)
+                        .send(withGodToken())
+                        .expect(HTTPStatus.FORBIDDEN);
+                    expect(res.body).toHaveProperty("reason", ValidationReasons.OFFER_EDIT_PERIOD_OVER(grace_period_over_test_offer._id));
+                });
+
+            });
+
+            test("should allow editing offer with valid grace period", async () => {
+                await test_agent
+                    .post(`/offers/edit/${grace_period_valid_test_offer._id.toString()}`)
+                    .send(withGodToken())
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("should allow editing offer in the future", async () => {
+                await test_agent
+                    .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                    .send(withGodToken())
+                    .expect(HTTPStatus.OK);
+            });
+
+            describe("testing edit dates in the past", () => {
+                test("should fail if publishDate in the past", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({ "publishDate": "2019-02-01" }))
+                        .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                    expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.DATE_EXPIRED);
```

Comment:
assert the field that is throwing the error as well  (i.e. check the `param` field of `res.body.errors[0]` as well as the `msg`)